### DISCUSSION TITLE
id_token not present in TokenSet

### DISCUSSION THREAD

**Discussion by jimfilippou** [nextauthjs/next-auth#3555] [Help]
Hello I am trying to get Okta provider to work with next-auth but I am getting this issue on the callback step.

https://next-auth.js.org/errors#oauth_callback_error **id_token not present in TokenSet**

My next-auth configuration looks like this 

```typescript
const options: NextAuthOptions = {
    providers: [
        OktaProvider({
            clientId: process.env.OKTA_CLIENTID,
            clientSecret: process.env.OKTA_CLIENTSECRET,
            issuer: process.env.OKTA_ISSUER,
        })
    ],
    theme: {
        colorScheme: "auto",
        brandColor: ""
        logo: ""
    },
    pages: {
        signIn: '/auth/login',
        error: '/auth/error'
    },
    secret: "secret",
}

const auth = (req, res) => NextAuth(req, res, options);

export default auth;
```

Anyone know what would be the cause? Thanks for your time and happy new year! 🥳

**Comment by balazsorban44** [comment]
@bogdansoare changed the default config to assume there is an id_token present in #3418, but it sounds like something might need to be configured in Okta. @bogdansoare any ideas?

**Reply by jimfilippou** [reply]
Doing a quick google search I came across this page https://community.auth0.com/t/id-token-not-present-in-tokenset/63690. There is an issue with scope in the above link, I believe we need to explicitly provide the scope of claims. Looking forward for a response from @bogdansoare

**Reply by balazsorban44** [reply]
Here are the default scopes:

https://github.com/nextauthjs/next-auth/blob/c71cb8457da0febe5d71283e9361014dda652c31/src/providers/okta.ts#L45

Which one is missing?

You also link to an Auth0 forum, but you mention the Okta provider. Am I missing something? :thinking:

**Reply by jimfilippou** [reply]
Doesn't look like something's missing. Also I want to add that authentication works when I do a refresh or manually go to a protected URL. As it seems, the only issue is the callback

**Reply by bogdansoare** [reply]
@jimfilippou I didn't do any additional configuration in Okta, the default scopes worked out of the box

<img width="536" alt="Screenshot 2022-01-04 at 14 05 26" src="https://user-images.githubusercontent.com/2750170/148056582-2b8f26c1-fc54-4bfb-879c-b95c646cb8a0.png">

**Reply by paulv317** [reply]
I am having an issue with the callback when I specifically do not send scope as a parameter to Okta. I have Okta set up to handle a request without the scopes parameter, and it should then look at the Group a user is in, and return the scopes for that user in the jwt. Per Okta documentation this should be allowed: https://developer.okta.com/docs/guides/customize-authz-server/main/#create-scopes - subheading #6. Any thoughts @balazsorban44 ?

**Reply by paulv317** [reply]
https://next-auth.js.org/errors#oauth_callback_error id_token not present in TokenSet {
  error: [Error [OAuthCallbackError]: id_token not present in TokenSet] {
    code: undefined
  },
  providerId: 'okta',
  message: 'id_token not present in TokenSet'
}

This is the error I am getting.

**Comment by robertgiannini** [comment]
I was getting this issue due to the scope param when attempting to login with Google. In a previous version of Next.js, you were supposed to pass in the scopes as a string separated by spaces, but in the lastest version of Next.js (as of this comment) you are supposed to pass an _array_ of scope strings. This behavior was cause by an update to my Next.js version.

**Reply by peter-olom** [reply]
I arrived at this problem just by updating next-auth. I remained on version 4.0.6 for a long time now. After seeing your response, I deleted the authorization prop which contained my scope and I was fine afterwards. I used the AzureAD provider. Thanks

**Comment by achinth** [comment]
Any solution to fix this issue?

**Comment by matheuslemke** [comment]
This worked for me:

              Google is an OIDC provider, make sure you include `openid` in scopes.

_Originally posted by @balazsorban44 in https://github.com/nextauthjs/next-auth/issues/4295#issuecomment-1084717751_

**Reply by syedamanat** [reply]
Worked for me as well.
